### ISSUE TITLE
Docker Desktop version 4.25.0 WSL Integration not working

### ISSUE THREAD

**Issue by kerbe** [WhitewaterFoundry/Pengwin#779]
**Describe the bug**
Once I upgraded my Docker Desktop to version 4.25.0, docker bridge to Pengwin doesn't work anymore. When I try docker commands, I get only error like this:
```
$ docker ps
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

When I downgraded back to Docker Desktop version 4.24.2, which is previous version, docker bridge works again.

While I had 4.25.0 installed, if I did `ps axu | grep -i docker`, that didn't show any processes running.

If I tried using Docker directly in Windows, all worked as expected, both via GUI and via command line commands in PowerShell, so it wasn't broken Docker Desktop installation.

**Basic Troubleshooting Checklist**

[x] I have searched Google for the error message.
[x] I have checked official WSL troubleshooting documentation: https://docs.microsoft.com/en-us/windows/wsl/troubleshooting#confirm-wsl-is-enabled.
[x] I have searched the official Microsoft WSL issues page: https://github.com/Microsoft/WSL/issues.
[x] I have searched the Pengwin issues page: https://github.com/WhitewaterFoundry/Pengwin/issues.
[x] I have run Windows 10 updates and restarted.
[x] I have restarted WSL service: `wsl.exe --terminate WLinux` `wsl.exe --shutdown`.

What other troubleshooting have you attempted?
I tried to run pengwin-setup again, and re-enable Docker bridge, but that didn't change anything.

Insert here:

**Pengwin Version**

Run `grep "PENGWIN_VERSION" /etc/os-release` in Pengwin.

Insert here:
PENGWIN_VERSION="23.07.1"

**WSL general info**

Run `wslsys` in Pengwin.

Insert here:
$ wslsys
Release Install Date: Fri Feb 10 18:41:30 EET 2023
Branch: ni_release
Build: 22621
Full Build: 22621.1.amd64fre.ni_release.220506-1250
Display Scaling: 1
Locale (Windows): en_US
Theme (Windows): dark
Uptime (Windows): 0d 4h 36m
Version (WSL): 2
Uptime (WSL): 0d 4h 35m
Release: Pengwin
Kernel: Linux 5.15.90.1-microsoft-standard-WSL2
Packages: 509
IPv4 Address: 172.18.9.102
System Type (Windows): Desktop
SystemD Status: disabled

**Comment by crramirez** [comment]
Hello @kerbe ,

Thanks a lot for reporting. Are you using Docker Desktop with Hyper-V backend or WSL2 one?

Regards,
Carlos

**Comment by kerbe** [comment]
I am using Docker Desktop with WSL2 backend.

**Comment by kerbe** [comment]
Alright, I have to apologize now. I have no idea how I have been this bad both searching for other issues similar to this, but being able to mix copy/pasting what I found. Due all that it looks like it isn't Pengwin problem, but Docker Desktop problem.

Last night when I posted previous comment, I had found this issue:
https://github.com/docker/for-win/issues/13774

When I started edit my comment here to include that link, I somehow managed to find issue 13764, and happily copy/paste that... without reading that, and realizing that was pretty much describing this problem I have (though I didn't see relation to Docker powersaver mode going on/off and that resuming WSL commands... they never came back for me in my testing).

So this issue doesn't most likely need actions from Pengwin developers, unless that issue in Docker Desktop repo brings some light what all WSL distros should do to continue using docker command bridge.

My deepest apologies for all extra work I have caused @crramirez.

I'll leave this issue open, so if other Pengwin users encounter same problem, they hopefully get to root cause of problem more easily, and this can be closed once upstream has released fix.

**Comment by crramirez** [comment]
No worries @kerbe. Have you tried as a temporal workaround to enable the communication through TCP?
![image](https://github.com/WhitewaterFoundry/Pengwin/assets/8397379/1c598274-a854-4ca1-9fb8-2a56c72704f8)

**Comment by kerbe** [comment]
Tried enabling communication through TCP, but enabling that alone didn't fix things.

I tested with Ubuntu WSL installation, and it seemed to be working just fine, without TCP communication enabled as well. This is somewhat strange. Issue in Docker repo seems to have concluded that some modifications in .bashrc / other login scripts can cause this problem, or using some other shell (csh, fish etc) than bash. Potentially it is then something in Pengwins login scripts, or something that I have configured myself for those, as Ubuntu without modifications works fine. 

As there is test build already done that seems to fix this in Dockers repo, https://github.com/docker/for-win/issues/13764#issuecomment-1798242872 I think it is enough to just notice that 4.25.0 is somehow broken. If others encounter same problem, downgrading is one option, using that test build is second option (I installed it today, and all seems to work as expected), or waiting until next official version is released.
